f/frizzle/comapeo/comapeo_observations.py

```diff
@@ -232,28 +262,27 @@ def download_and_transform_comapeo_data(
             logger.info("Response received: ", response.text)
             raise ValueError("Invalid JSON response received from server.")
 
-        for observation in current_project_data:
+        for i, observation in enumerate(current_project_data):
             observation["project_name"] = project_name
             observation["project_id"] = project_id
 
-            # Convert keys from camelCase to snake_case
-            observation = {
-                ("docId" if k == "docId" else camel_to_snake(k)): v
-                for k, v in observation.items()
-            }
+            # Create k/v pairs for each tag
+            if "tags" in observation:
+                for key, value in observation["tags"].items():
+                    observation[key] = value
+                del observation["tags"]
+
+            # Convert keys from camelCase to snake_case, handling key collisions and char limits
+            observation = normalize_and_snakecase_keys(observation)
```

Comment:
Previously we special-cased `docId` right here, never transforming it. That way the key was sure to be as-expected when we check it in line 407 (of the old code).

By making this change, we risk `docId` being a key that gets modified by `normalize_and_snakecase_keys`:

```py

observation = {"DOC-id": "phony", "docId": "the-real-one"}
normalize_and_snakecase_keys(observation)
>>> {'doc_id': 'phony', 'doc_id_2': 'the-real-one'}
```

Reply:
Good catch, thanks. Opted for adding a special-case param inside `normalize_and_snakecase_keys`.